modules/listener/listener.bal

```diff
@@ -0,0 +1,154 @@
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerinax/googleapis_drive as drive;
+
+# Listener Configuration. 
+#
+# + port - Port for the listener.  
+# + specificFolderOrFileId - Folder or file Id.  
+# + callbackURL - Callback URL registered.   
+# + clientConfiguration - Drive client connecter configuration.  
+# + eventService - 'OnEventService' object with supported events. 
+public type ListenerConfiguration record {
+    int port;
+    string callbackURL;
+    drive:Configuration clientConfiguration;
+    OnEventService eventService;
+    string? specificFolderOrFileId = ();
+};
+
+# Drive event listener   
+@display {label: "Google Drive Listener"}
+public class DriveEventListener {
+    private http:Listener httpListener;
+    private string currentToken;
+    private string channelUuid;
+    private string watchResourceId;
+    private http:Client clientEP;
+    private OnEventService eventService;
+    private json[] currentFileStatus = [];
+    private string specificFolderOrFileId;
+    private drive:Client driveClient;
+    private drive:WatchResponse watchResponse;
+    private boolean isWatchOnSpecificResource = false;
+    private boolean isAFolder = true;
+
+    # Listener initialization
+    public function init(ListenerConfiguration config) returns @tainted error? {
+        self.eventService = config.eventService;
+        self.httpListener = check new (config.port);
+        self.driveClient = check new (config.clientConfiguration);
+        if (config.specificFolderOrFileId is string) {
+            self.isAFolder = check checkMimeType(self.driveClient, config.specificFolderOrFileId.toString());
+        }
+        if (config.specificFolderOrFileId is string && self.isAFolder == true) {
+            check validateSpecificFolderExsistence(config.specificFolderOrFileId.toString(), self.driveClient);
+            self.specificFolderOrFileId = config.specificFolderOrFileId.toString();
+            self.watchResponse = check startWatch(config.callbackURL, self.driveClient, self.specificFolderOrFileId.
+            toString());
+            check getCurrentStatusOfDrive(self.driveClient, self.currentFileStatus, 
+            self.specificFolderOrFileId.toString());
+            self.isWatchOnSpecificResource = true;
+        } else if (config.specificFolderOrFileId is string && self.isAFolder == false) {
```

Comment:
Same here. shall we use !self.isAFolder?

Reply:
Hi
Renamed to isFolder in all occurrences.
Thank you